src/main/java/io/trino/plugin/tiledb/TileDBRecordCursor.java

```diff
@@ -447,11 +462,120 @@ private void setRanges(TileDBSplit split) throws TileDBError
                 dimIdx++;
                 dimension.close();
             }
+            //iterate attributes
+            HashMap<String, Attribute> attributes = arraySchema.getAttributes();
+            Iterator it = attributes.entrySet().iterator();
+            while (it.hasNext()) {
+                Map.Entry pair = (Map.Entry) it.next();
+                Attribute att = (Attribute) pair.getValue();
+                Pair attBounds = getBoundsForAttribute(split, att);
+                Object lowBound;
+                Object highBound;
+                boolean isVar = att.isVar();
+                if ((isVar && attBounds.getFirst() != null && attBounds.getFirst().toString().equals(""))) { //empty string case
+                    QueryCondition cond;
+                    if (att.getNullable()) {
+                        cond = new QueryCondition(ctx, att.getName(), "".getBytes(), att.getType().javaClass(), TILEDB_EQ);
+                    }
+                    else {
+                        cond = new QueryCondition(ctx, att.getName(), " ".getBytes(), att.getType().javaClass(), TILEDB_EQ);
+                    }
+                    query.setCondition(cond);
+                    continue;
+                }
+                if ((attBounds.getFirst() != null || attBounds.getSecond() != null)) {
```

Comment:
Can we abstract the conditions here to reduce the duplication? eg maybe a local lambda?

```
BiFunction<Object, boolean, Object> getCond = (Object o, boolean isVar) -> {
  ...
}
var lowBound = getBound(attBounds.getFirst(), isVar);
var highBound = getBound(attBounds.getSecond(), isVar);
QueryCondition cond1 = ...
```
(or maybe `isVar` can be captured)